### PR TITLE
TL/UCP: reduce_scatterV ring algorithm

### PR DESCRIPTION
**Pull request by vspetrov**
## What
Implements ReduceScatterV ring algorithm in TL/UCP
Adds RSV to test/mpi and gtest

### CODE REVIEW

File: src/components/tl/ucp/tl_ucp_coll.h [hunk 1]
```diff
@@ -65,6 +65,14 @@ typedef struct ucc_tl_ucp_task {
...
             char                    s_scratch_busy[2];
         } reduce_scatter_ring;
+        struct {
+            void *       scratch;
+            size_t       max_block_count;
+            ucc_ep_map_t inv_map;
+            int          n_frags;
+            int          frag;
+            char         s_scratch_busy[2];
+        } reduce_scatterv_ring;
```
Comment (Sergei-Lebedev) [inline src/components/tl/ucp/tl_ucp_coll.h:75]: alignment

File: test/mpi/test_mpi.cc [hunk 2]
```diff
@@ -298,6 +298,7 @@ int ucc_coll_is_rooted(ucc_coll_type_t c)
     case UCC_COLL_TYPE_ALLTOALLV:
     case UCC_COLL_TYPE_BARRIER:
     case UCC_COLL_TYPE_REDUCE_SCATTER:
+    case UCC_COLL_TYPE_REDUCE_SCATTERV:
```
Comment (Sergei-Lebedev) [inline test/mpi/test_mpi.cc:301]: need to add UCC_COLL_TYPE_REDUCE_SCATTERV to run_all_at_team to run different datatypes and ops

**Review by Sergei-Lebedev — APPROVED** [review]

File: test/mpi/test_mpi.cc (hunk 2 above)
Comment (vspetrov) [reply test/mpi/test_mpi.cc:301]: done

File: src/components/tl/ucp/reduce_scatterv/reduce_scatterv.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021-2022.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+#ifndef REDUCE_SCATTERV_H_
+#define REDUCE_SCATTERV_H_
+#include "../tl_ucp_reduce.h"
```
Comment (bureddy) [inline src/components/tl/ucp/reduce_scatterv/reduce_scatterv.h:8]: isn't it src/componenets/tl/ucp in include path? do we need relative path for this header?

File: src/components/tl/ucp/reduce_scatterv/reduce_scatterv_ring.c [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+    ucc_tl_ucp_put_schedule(&schedule->super.super);
+    return status;
+}
+
+ucc_status_t
+ucc_tl_ucp_reduce_scatterv_ring_init(ucc_base_coll_args_t *coll_args,
+                                     ucc_base_team_t *     team,
+                                     ucc_coll_task_t **    task_h)
+{
+
```
Comment (bureddy) [inline src/components/tl/ucp/reduce_scatterv/reduce_scatterv_ring.c:346]: empty line

File: src/components/tl/ucp/reduce_scatterv/reduce_scatterv_ring.c [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+    max_segcount = ucc_coll_args_get_max_count(
+        &coll_args->args, coll_args->args.dst.info_v.counts, size);
+    /* in flight we can have 2 sends from 2 differnt blocks and 1 recv:
+       need 3 * max_segcount of scratch per set */
+    count_per_set = (max_segcount + n_subsets - 1) / n_subsets;
+
+    to_alloc_per_set = count_per_set * 3;
+    status           = ucc_mc_alloc(&tl_schedule->scratch_mc_header,
+                          to_alloc_per_set * dt_size * n_subsets, mem_type);
+
```
Comment (bureddy) [inline src/components/tl/ucp/reduce_scatterv/reduce_scatterv_ring.c:396]: empty line

**Review by bureddy — APPROVED** [review]

File: src/components/tl/ucp/reduce_scatterv/reduce_scatterv.h (hunk 3 above)
Comment (vspetrov) [reply src/components/tl/ucp/reduce_scatterv/reduce_scatterv.h:8]: we don't need relative indeed since we build the whole TL from tl/ucp folder

**Pull request merged**